File: 总题库/150.EvaluateReversePolishNotation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for token in tokens:
            if token[0].isdigit() or (len(token) > 1 and token[1].isdigit()):
                st.append(int(token))
            else:
                num2 = st[-1]
                st.pop();
                num1 = st[-1]
                st.pop();
                if token == '+':
                    num3 = num1 + num2
                if token == '-':
                    num3 = num1 - num2
                if token == '*':
                    num3 = num1 * num2
                if token == '/':
                    num3 = int(num1 / num2)
                st.append(num3)
        return st[-1]

File: 总题库/150.EvaluateReversePolishNotation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for (string token : tokens) {
            if (isdigit(token[0])) {
                st.push(stoi(token));
            }
            else {
                int num2 = st.top();
                st.pop();
                int num1 = st.top();
                st.pop();
                int num3;
                if (token == "+") num3 = num1 + num2;
                if (token == "-") num3 = num1 - num2;
                if (token == "*") num3 = num1 * num2;
                if (token == "/") num3 = num1 / num2;
                st.push(num3); 
            }
        }
        return st.top();
    }
};